nameresolution/structuredformat/structuredformat.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package structuredformat
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	yaml "gopkg.in/yaml.v3"
+
+	"github.com/dapr/components-contrib/metadata"
+	nr "github.com/dapr/components-contrib/nameresolution"
+	"github.com/dapr/kit/logger"
+	kitmd "github.com/dapr/kit/metadata"
+)
+
+const (
+	JSONStringStructuredValue = "jsonString"
+	YAMLStringStructuredValue = "yamlString"
+	JSONFileStructuredValue   = "jsonFile"
+	YAMLFileStructuredValue   = "yamlFile"
+)
+
+var allowedStructuredTypes = []string{
+	JSONStringStructuredValue,
+	YAMLStringStructuredValue, JSONFileStructuredValue, YAMLFileStructuredValue,
+}
+
+// StructuredFormatResolver parses service names from a structured string
+// defined in the configuration.
+type StructuredFormatResolver struct {
+	meta      structuredFormatMetadata
+	instances appInstances
+
+	logger logger.Logger
+}
+
+// structuredFormatMetadata represents the structured string (such as JSON or YAML)
+// provided in the configuration for name resolution.
+type structuredFormatMetadata struct {
+	StructuredType string
+	StringValue    string
+	FilePath       string
+}
```

Comment:
can you pls add a metadata.yaml file denoting which fields are required, defaults set, etc. You can reference the other components metadata.yaml files for this :)